cooperator/models/subscription_request.py

```diff
@@ -85,55 +80,59 @@ def _check_share_available_to_user(self):
     def _send_confirmation_mail(self):
         if self.company_id.send_confirmation_email:
             mail_template_notif = self.get_mail_template_notif(
-                is_company=self.partner_id.is_company
+                is_company=self.is_company
             )
             # sudo is needed to change state of invoice linked to a request
             #  sent through the api
             mail_template_notif.sudo().send_mail(self.id)
 
-    # todo code this in a more pythonic way
-    #  - do not bypass self.create in create_comp_sub_req
-    #  - use "not in" where applicable
-    #  - sanitize declaration and assignation of cooperator
-    #  beware if usage in investor_wallet_platform
+    def _find_partner_from_create_vals(self, vals):
+        """
+        Find the partner corresponding to the vals dict.
+
+        If partner_id is present in the dict, use it to find the partner.
+        Otherwise, search companies by register number and individuals by
+        email address. If a match is found, add partner_id in the dict.
+        """
+        partner_model = self.env["res.partner"]
+        partner_id = vals.get("partner_id")
+        if partner_id:
+            return partner_model.browse(partner_id)
+        if vals.get("is_company"):
+            partner = partner_model.get_cooperator_from_crn(
+                vals.get("company_register_number")
+            )
+        else:
+            partner = partner_model.get_cooperator_from_email(vals.get("email"))
+        if partner:
+            vals["partner_id"] = partner.id
```

Comment:
Don't you want to return a partner in this case as well ?

And if nothing is found, isn't it better to return an empty `RecordSet` rather than an implicit `None`

Reply:
my bad, thanks!